src/cdk-experimental/scrolling/virtual-scroll-viewport.spec.ts

```diff
@@ -339,6 +325,132 @@ describe('CdkVirtualScrollViewport', () => {
                 `rendered content size should match expected value at offset ${offset}`);
       }
     }));
+
+    it('should work with an Observable', fakeAsync(() => {
+      const data = new Subject<number[]>();
+      testComponent.items = data as any;
+      finishInit(fixture);
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 0}, 'no items should be rendered');
+
+      data.next([1, 2, 3]);
+      fixture.detectChanges();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 3}, 'newly emitted items should be rendered');
+    }));
+
+    it('should work with a DataSource', fakeAsync(() => {
+      const data = new Subject<number[]>();
+      testComponent.items = new ArrayDataSource(data) as any;
+      finishInit(fixture);
+      flush();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 0}, 'no items should be rendered');
+
+      data.next([1, 2, 3]);
+      fixture.detectChanges();
+      flush();
+
+      expect(viewport.getRenderedRange())
+          .toEqual({start: 0, end: 3}, 'newly emitted items should be rendered');
+    }));
+
+    it('should trackBy value', fakeAsync(() => {
```

Comment:
Is trackBy value the default?

Reply:
Yes I think so, due to the way the setter works I can't actually pass undefined though. I should probably just make that work.